src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
+                ],
+                "name_original": calced_value,
+            }
+            for (calced_value, calcs) in calcs.items()
+        }
+        names_non_calcs = [
+            name for name in xbrl_meta_tbl.xbrl_factoid if name not in calced_values
+        ]
+        non_calcs = {
+            self.rename_column_xbrl_to_pudl(name): {"name_original": name}
+            for name in names_non_calcs
+        }
+        return calcs_renamed | non_calcs
+
+    def rename_column_xbrl_to_pudl(
+        self,
+        col_name_xbrl: str,
+    ) -> str:
+        """Rename a column name from orignal XBRL name to the transformed PUDL name.
+
+        There are several transform params that either explicitly or implicity rename
+        columns:
+        * :class:`RenameColumnsFerc1`
+        * :class:`WideToTidySourceFerc1`
+        * :class:`UnstackBalancesToReportYearInstantXbrl`
+        * :class:`ConvertUnits`
+
+        This method attempts to use the table params to translate a column name.
+
+        Note: Instead of doing this for each individual column name, we could compile a
+        rename dict for the whole table with a similar processand then apply it for each
+        group of columns instead of running through this full process every time. If
+        this took longer than...  ~5 ms on a single table w/ lots of calcs this would
+        probably be worth it for simplicity.
+        """
+        rename_dicts = [
+            self.params.rename_columns_ferc1.duration_xbrl,
+            self.params.rename_columns_ferc1.instant_xbrl,
+            self.params.rename_columns_ferc1.xbrl,
+        ]
+        col_name_new = col_name_xbrl
+        for rename_stage in rename_dicts:
+            col_name_new = str(rename_stage.columns.get(col_name_new, col_name_new))
+        # the values in wide_to_tidy are found at the end of the column names and
+        # extracted, so we need to remove all of the suffixes.
+        wide_to_tidy_value_types = []
+        for wide_to_tidy_stage in json.loads(self.params.wide_to_tidy.json()).values():
+            if not isinstance(wide_to_tidy_stage, list):
+                wide_to_tidy_stage = [wide_to_tidy_stage]
+            for wide_to_tidy in wide_to_tidy_stage:
+                if wide_to_tidy["value_types"]:
+                    wide_to_tidy_value_types.append(wide_to_tidy["value_types"])
+        wide_to_tidy_value_types = pudl.helpers.dedupe_n_flatten_list_of_lists(
+            wide_to_tidy_value_types
+        )
+
+        for value_type in wide_to_tidy_value_types:
+            if col_name_new.endswith(f"_{value_type}"):
+                col_name_new = re.sub(f"_{value_type}$", "", col_name_new)
+            # col_name_new = col_name_new.replace(f"_{value_type}$", "")
+
+        if self.params.unstack_balances_to_report_year_instant_xbrl:
+            # TODO: do something...? add starting_balance & ending_balance suffixes?
+            pass
+        if self.params.convert_units:
```

Comment:
Seems like this will be uncommon, but there are some tables that have a mix of money and other things like customers or MWh, so it might be necessary. Also if this method lives in the `Ferc1TableTransformParams` class it should work for the non-money tables too.